cleanup/artifactCleanup/ArtifactCleanupTest.groovy

```diff
@@ -1,41 +1,112 @@
 
         def builder = artifactory.repositories().builders()
-        def repository = builder.localRepositoryBuilder().key('maven-local')
-        .repositorySettings(new MavenRepositorySettingsImpl()).build()
+        def repository = builder.localRepositoryBuilder().key(repoName)
+            .repositorySettings(new MavenRepositorySettingsImpl()).build()
         artifactory.repositories().create(0, repository)
 
+        return artifactory
+    }
+
+    def 'artifact cleanup test'() {
+        setup:
+        def artifactory = createClientAndMavenRepo('maven-local')
+
         def file = new ByteArrayInputStream('test'.bytes)
-        artifactory.repository("maven-local").upload('test', file).doUpload()
+        artifactory.repository('maven-local').upload('test', file).doUpload()
 
         when:
         artifactory.plugins().execute('cleanup').
             withParameter('repos', 'maven-local').sync()
-        artifactory.repository("maven-local").file('test').info()
+        artifactory.repository('maven-local').file('test').info()
+
+        then:
+        notThrown(HttpResponseException)
+
+        when:
+        artifactory.plugins().execute('cleanup').
+            withParameter('repos', 'maven-local').
+            withParameter('months', '0').sync()
+        artifactory.repository('maven-local').file('test').info()
+
+        then:
+        thrown(HttpResponseException)
+
+        cleanup:
+        artifactory.repository('maven-local').delete()
+    }
+
+    def 'artifact cleanup skip artifact test'() {
+        setup:
+        def artifactory = createClientAndMavenRepo('maven-local')
+
+        def file = new ByteArrayInputStream('test'.bytes)
+        artifactory.repository('maven-local').upload('test', file).withProperty('cleanup.skip', 'true').doUpload()
+
+        when:
+        artifactory.plugins().execute('cleanup').
+            withParameter('repos', 'maven-local').
+            withParameter('months', '0').sync()
+        artifactory.repository('maven-local').file('test').info()
 
         then:
         notThrown(HttpResponseException)
 
         when:
         artifactory.plugins().execute('cleanup').
             withParameter('repos', 'maven-local').
+            withParameter('months', '0').
+            withParameter('disablePropertiesSupport', 'true').sync()
+        artifactory.repository('maven-local').file('test').info()
+
+        then:
+        thrown(HttpResponseException)
+
+        cleanup:
+        artifactory.repository('maven-local').delete()
+    }
+
+    def 'artifact cleanup skip folder test'() {
+        setup:
+        def artifactory = createClientAndMavenRepo('maven-local')
+
+        def file = new ByteArrayInputStream('test'.bytes)
+        artifactory.repository('maven-local').upload('foo/bar/test', file).withProperty('cleanup.skip', 'true').doUpload()
+        artifactory.repository('maven-local').folder('foo').properties().addProperty('cleanup.skip', 'true').doSet();
```

Comment:
@axel3rd I believe it is important to have one test for each level in the hierarchy with the property set. With this test as it is, it is difficult to tell which property is causing the artifact to be skipped in the clean up process. What is the behavior when we set the property at the artifact's parent folder only? What about when we set the property at the parent's parent folder only?